pkg/plugins/runtime/k8s/controllers/mesh_defaults_controller.go

```diff
@@ -0,0 +1,63 @@
+package controllers
+
+import (
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	kube_core "k8s.io/api/core/v1"
+	kube_ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	"github.com/kumahq/kuma/pkg/core/resources/manager"
+	"github.com/kumahq/kuma/pkg/core/resources/model"
+	defaults_mesh "github.com/kumahq/kuma/pkg/defaults/mesh"
+	mesh_k8s "github.com/kumahq/kuma/pkg/plugins/resources/k8s/native/api/v1alpha1"
+)
+
+// MeshDefaultsReconciler creates default resources for a created Mesh
+type MeshDefaultsReconciler struct {
+	Log             logr.Logger
+	ResourceManager manager.ResourceManager
+}
+
+func (r *MeshDefaultsReconciler) Reconcile(req kube_ctrl.Request) (kube_ctrl.Result, error) {
+	log := r.Log.WithValues("mesh", req.NamespacedName)
+
+	// we skip default name Mesh because pkg/defaults/mesh.go uses ResourceManager that executes CreateDefaultMeshResources already
+	if req.Name == model.DefaultMesh {
+		return kube_ctrl.Result{}, nil
+	}
+
+	if err := defaults_mesh.CreateDefaultMeshResources(r.ResourceManager, req.Name); err != nil {
+		log.Error(err, "could not create default mesh resources")
```

Comment:
What if it's gonna fail here? If I understand it right it reconciles only on Create? Probably we need to return here:
```go
return kube_ctrl.Result{
    Requeue: true,
}, err
```
to run it again?

Reply:
That's a good point. I should return here an error. Then Kubernetes does the retry with exponential backoff for us.

Reply:
(I tested it, even with create only - it does the retry)